product_pricelist_revision/models/pricelist.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2018 Carlos Dauden - Tecnativa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = 'product.pricelist.item'
+
+    name = fields.Char(search='_search_name')
+
+    previous_item_id = fields.Many2one(
+        comodel_name='product.pricelist.item',
+        string='Previous Item',
+        help='Relation with previous item when duplicate line',
+    )
+    previous_price = fields.Float(
+        related='previous_item_id.fixed_price',
+        string='Previous Fixed Price',
+        readonly=True,
+    )
+    variation_percent = fields.Float(
+        compute='_compute_variation_percent',
+        store=True,
+        digits=dp.get_precision('Product Price'),
+        string='Variation %',
+    )
+
+    @api.model
+    def _search_name(self, operator, value):
+        def name_matching(item):
```

Comment:
Can you try this?
```suggestion
        return [
            '|', '|',
            ('categ_id', operator, value),
            ('product_tmpl_id', operator, value),
            ('product_id', operator, value),
        ]
```
It is not as "perfect" but it is more efficient

Reply:
But why do we need to overwrite this?

Reply:
Name is a computed field without search:
https://github.com/odoo/odoo/blob/35d4a8ccce8f3a34701d231e17a324cbbf5abfab/addons/product/models/product_pricelist.py#L457-L459

Reply:
> Can you try this?
> 
> It is not as "perfect" but it is more efficient.
> 
> Based on name generation method:
> https://github.com/odoo/odoo/blob/35d4a8ccce8f3a34701d231e17a324cbbf5abfab/addons/product/models/product_pricelist.py#L480-L487

I think it could be a good balance, do you agree @pedrobaeza ?

Reply:
Yeah, it's OK.